storybookjs/storybook
# UI: Add the new structure for the mobile layout

Closes #24213 

## What I did

This PR is based on all the work done here https://github.com/storybookjs/storybook/pull/20412. In this PR we are focusing on creating the base layout for mobile and allocating a spot on mobile for the main navigation. The content of this new section will be created in following PRs. Here's a list of the work done here:
- Update `useMediaQuery` (initially created by @valentinpalkovic) to use `window.matchMedia` instead
- Hide sidebar and addon panel on Mobile
- Adjust the layout and preview sizing to work on mobile
- Add the core structure for the mobile navigation

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

To test it locally, open Storybook UI library from the monorepo with `yarn storybook:ui` and test that you can see all stories correctly on mobile. You should also see an empty bar at the bottom for the future mobile navigation.

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
